File: ECommerceAPI/Models/Offer.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace ECommerceAPI.Models
{
    public partial class Offer
    {
        public Offer()
        {
            OrderItems = new HashSet<OrderItem>();
            Orders = new HashSet<Order>();
        }

        public int OfferId { get; set; }
        public string OfferName { get; set; }
        public string OfferDescription { get; set; }
        public int? ProductId { get; set; }
        public int? DiscountPercentage { get; set; }
        public double? DiscountAmount { get; set; }
        public bool? IsActive { get; set; }
        public DateTime? CreatedOn { get; set; }
        public DateTime? ModifiedOn { get; set; }
        public bool? IsDeleted { get; set; }
        public DateTime? DeletedOn { get; set; }

        public virtual Product Product { get; set; }
        public virtual ICollection<OrderItem> OrderItems { get; set; }
        public virtual ICollection<Order> Orders { get; set; }
    }
}

File: ECommerce-Bill_Feature/src/app/models/Offer.ts
import { Product } from "./Product";
import { Cart } from "./Cart";
import { OrderItem } from "./OrderItem";
import { Order } from "./Order";

export class Offer {
    offerId: number;
    offerName: string;
    offerDescription: string;
    productId: number | null;
    discountPercentage: number | null;
    discountAmount: number | null;
    isActive: boolean | null;
    createdOn: string | null;
    modifiedOn: string | null;
    isDeleted: boolean | null;
    deletedOn: string | null;
    product: Product;
    carts: Cart[];
    orderItems: OrderItem[];
    orders: Order[];
}